sidecar/app/query/base.py

```diff
@@ -77,23 +83,55 @@ def get_from_query_id(cls, query_id) -> Optional["Query"]:
             if query:
                 return query
             raise e
-        if query.status_file_path.exists():
-            with query.status_file_path.open("r") as f:
-                status_str = f.readline()
-                query.status = Status[status_str]
-                return query
-        return None
+        query.load_history_from_file()
+        if query.status == Status.UNKNOWN:
+            return None
+        return query
+
+    def load_history_from_file(self):
```

Comment:
I would suggest to add a comment showing an example how log file looks like, so we can compare and check that it is doing the right thing

Reply:
there's also a test that does this in the next PR